### PR TITLE
Add demodulize call to name to help with models inside modules extensions

### PR DESCRIPTION
**Pull request by Agroang**
Initially this PR wanted only to tackle problems with extensions for models that are inside modules as they did not get their extensions included as their file names do not match the expected paths due to the name space. This can be solved with the current version but you need to add the extension files inside the module named folders and extra module name spaces for the extensions, sometimes repeating two times the same name space to just make it work, which make no sense.

To solve this we can call `demodulize` on the `name` variable on the `Extensions` file to get the correct path.

Nevertheless, to be able to make the change I had to update the current CI actions and some dependencies as the current actions were deprecated and some of the dummy apps still being used for testing have already reached their EOL. Lots of the line changes are dependency updates for the rails `6.0` dummy app.

The workflows were changed according to the deprecation warnings recommendations.

Example of the faced problem:
`app/models/online/product.rb`

```rb
Module Online
  Class Product < ApplicationRecord
  
    def price
     # ...
    end
  end
end
```

Extension that we want to add:
`drivers/online/extensions/product_extension.rb`

```rb
module Online
  # Extends Online::Product model with some Online  specific behavior
  module ProductExtension
    extend ActiveSupport::Concern

    included do
      has_many :online_reviews, class_name: 'Online::Review',
                         foreign_key: :online_review_id
    end
  end
end
```

Without the proposed changes, this was not working, and the only way to make it work would be doing something like:
Create an extra folder for the extension to represent the name space:
`drivers/online/extensions/online/product_extension.rb`

And wrap the extension in another layer of the same module (as that seems to be the path that is trying to find):

```rb
module Online
  module Online
    # Extends Online::Product model with some Online  specific behavior
    module ProductExtension
      extend ActiveSupport::Concern

      included do
        has_many :online_reviews, class_name: 'Online::review',
                           foreign_key: :online_review_id
      end
    end
  end
end
```

With the proposed changes, the first version should work without having to add extra folders and extra wrappings of the extensions for named spaced models.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -25,22 +25,21 @@ jobs:
     strategy:
       matrix:
         versions:
-        - rails: '5.2'
-          ruby: '2.7'
```
Comment (Agroang) [inline .github/workflows/main.yml:29]: ruby 2.7 reached its EOL

File: spec/support/dummy_app_helpers.rb [hunk 2]
```diff
@@ -119,6 +119,8 @@ def teardown_dummy_app
   private
 
   def cp_r(src, dst)
+    # Depending on your machine's file system you may need to change
+    # the following command into system "cp -R #{src} #{dst}"
```
Comment (Agroang) [inline spec/support/dummy_app_helpers.rb:123]: While trying to run the specs this line would break for macOS.

**Comment by degikko** [comment]
@FTLam11 can you help us review this PR, please?

**Comment by degikko** [comment]
@Resonious can you help us review this PR, please?

File: lib/rails_drivers/extensions.rb
Comment (FTLam11) [inline lib/rails_drivers/extensions.rb:1]: Maybe can mention this update in [changelog](https://github.com/degica/rails_drivers/blob/master/CHANGELOG.md)? Perhaps something like:

* Namespaced core classes can be enhanced without additionally namespacing extensions

I wonder if anyone is already nesting namespaces for their extensions, could be a breaking change 😨

File: .github/workflows/main.yml (hunk 1 above)
Comment (FTLam11) [reply .github/workflows/main.yml:29]: Outside scope of this PR, but could be nice to make a dummy app for Ruby 3.X/Rails 7 and run tests against them in CI

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -25,22 +25,21 @@ jobs:
...
+        uses: ruby/setup-ruby@v1
         with:
           ruby-version: ${{ matrix.versions.ruby }}
       - name: Install sqlite3
         run: sudo apt-get install -y libsqlite3-dev
       - name: Install nodejs dependencies
-        run: |
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.2'
```
[the remark this answers is not in the document]
Comment (FTLam11) [reply .github/workflows/main.yml:41]: https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported

**Review by FTLam11 — APPROVED** [review]
Just one suggestion to update the `CHANGELOG`, other than that L:zany_face:TM, thanks for this PR!

If we remove the dummy app for Rails 5.2, might also need to update [`gemspec`](https://github.com/degica/rails_drivers/blob/master/rails_drivers.gemspec#L24-L33).

**Review by Resonious — COMMENTED** [review]
Thanks for fixing the CI. The bit rot is real :sob: 

I may be missing something but it looks to me like this makes it impossible to write an extension for any namespaced model. For example, would this work?

```ruby
# app/models/session/line_item.rb

class Session::LineItem
  # ...
end
```

```ruby
# drivers/hosted_page/extensions/session/line_item_extension.rb

module HostedPage
  module Session::LineItemExtension
    extend ActiveSupport::Concern

    # ...
  end
end
```

To me, :point_up_2: looks like a reasonable use-case that might not work after this PR but do correct me if I'm wrong. Unfortunately I think `Online::Online::ProductExtension` might be the way to go :sweat_smile:

**Comment by Agroang** [comment]
I see... 🤔 , indeed, I don't think that this approach would work with that sort of nested `*_extension.rb` files. 
I will take a deeper look when I have a little bit of more time available to try to make it work for such cases.
Worst case scenario it may end up being just be a PR that updates CI and some dependencies for the dummy app 6.0 😅

**Comment by Agroang** [comment]
Totally forgot to close this one 🙇

**Pull request closed without merging**